### PR TITLE
fix(lib): polynomial redos routes

### PR DESCRIPTION
**Pull request by Fdawgs**
See https://github.com/fastify/fastify-autoload/pull/330

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/spec/openapi/utils.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ function normalizeUrl (url, serverUrls, stripBasePath) {
 
 function resolveServerUrls (servers) {
   const resolvedUrls = []
-  const findVariablesRegex = /{(.*?)}/g // As for OpenAPI v3 spec url variables are named in brackets, e.g. {foo}
+  const findVariablesRegex = /\{(.*?)\}/gu // As for OpenAPI v3 spec url variables are named in brackets, e.g. {foo}
```
Comment (gurgunday) [inline lib/spec/openapi/utils.js:91]: Was the escape necessary?

Just curious

Comment (Fdawgs) [reply lib/spec/openapi/utils.js:91]: Without it they are treated as quantifiers.

**Comment by mcollina** [comment]
@Fdawgs CodeQL is complaining about this regexp, could you take a look?

**Comment by Fdawgs** [comment]
> @Fdawgs CodeQL is complaining about this regexp, could you take a look?

This appears to have been an issue prior to this change (if you look in the Security tab of the repo).

**Comment by Fdawgs** [comment]
Found another one in https://github.com/fastify/fastify-swagger/pull/763/commits/698c66a2f9b1352f430bb8b4d2d86c7abbc5a9b5

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**